kopf/k8s/events.py

```diff
@@ -56,11 +64,15 @@ def post_event(*, obj, type, reason, message=''):
         event_time=now.isoformat() + 'Z',  # '2019-01-28T18:25:03.000000Z'
     )
 
+    api = kubernetes.client.CoreV1Api()
+    loop = asyncio.get_event_loop()
+    if not loop:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
```

Comment:
Also, since `post_event` is an async coroutine already, it runs in some event-loop. Therefore, `asyncio.get_running_loop()` would be sufficient and guaranteed to return an event-loop. And, [according to the docs](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop), it is recommended over `asyncio.get_event_loop()` because it is slightly faster/easier.

Reply:
As I understand, if you are already not in master thread, get_running_loop will raise RuntimeError. Same story with get_event_loop - it's for use in master thread, so, supposed you have only one event loop, and run all required tasks in it.
In current kopf architecture it is hard to create 1 event loop and run all kinds of tasks in it, so this is dirty hack to create new event loops inside functions that by themselves already run in slave threads.
It requires some controller object to be created before any of kopf functionality is initiated. So you could then create it, configure it, run/chain and control lifecycle of any required tasks(watching/queueing/handling/event-posting/object-patching) in it.
I understand that this current solution is dirty, but I don't know how to do it better at low costs right now.

Reply:
Reading it again. Can't really understand how it works from docs. Will try ```get_running_loop``` later and comment here on results.

Reply:
Changed to ```get_running_loop```. Seems to work fine, tests are passing.

Reply:
Thanks. 

To clarify: every thread has its own loop by default. `get_event_loop()` gets-or-creates it in the current thread. However, those are different event-loops for each thread, i.e. not the same loop as in the main thread.

`get_current_loop()` is only possible from inside a coroutine, as the coroutines already run in a loop (by definition). 

In case of Kopf, it will be always the main thread's event-loop (or a thread where the `run()` method is called), as other threads are used only for sync functions via the thread pools — unless we intentionally create and use a loop inside of those sync functions (but we don't).

Therefore, the effect of these lines is the same: it will be a main thread's event-loop. But a one-liner `get_current_loop()` is more simple and clear than the four-liner above.